R/nodeCentricSteinerTreeProblem-class.R

```diff
@@ -406,15 +420,21 @@ nodeCentricSteinerTreeProblem <- R6Class("nodeCentricSteinerTreeProblem",
           private$currentSolutionIndices <- which(solVec > 0)
         }
         
-        disconnectedComponentList <- decompose( self$getCurrentSolutionGraph() )
-        # The nodeDT table and inComponent variable keeps track of which node is where
+        disconnectedComponentList <- decompose( self$getCurrentSolutionGraph())# The nodeDT table and inComponent variable keeps track of which node is where
+        
         private$nodeDT[,inComponent := NA_integer_]
         
-        for(i in 1:length(disconnectedComponentList)){
+        
+        if(vcount(self$getCurrentSolutionGraph()) == 0){
+          
+          warning("No Solution was found, most probably the solver run out of time. Try an advanced MILP solver like CPLEX") 
+          
+          }else{
+        
+            for(i in 1:length(disconnectedComponentList)){
```

Comment:
This would trigger an error when the solver reached the time limit so I've added the lines above.

Reply:
Any error from cplex triggers this warnings even if it has nothing to do with time which isn't not the best. I'm going to check if there is a way to print the error message from cplex itself

Reply:
Worth opening a separate issue there I think